Sources/NIO/SocketChannel.swift

```diff
@@ -290,11 +290,7 @@ final class SocketChannel: BaseSocketChannel<Socket> {
             return
         }
 
-        guard let data = data.tryAsIOData() else {
-            promise?.fail(error: ChannelError.writeDataUnsupported)
```

Comment:
I think we should also remove `ChannelError.writeDataUnsupported` completely ?

Reply:
ACK, done.